app/services/creators/cash_transactions_creator.rb

```diff
@@ -0,0 +1,83 @@
+module Creators
+  class CashTransactionsCreator < BaseCreator
+    delegate :gross_income_summary, to: :assessment
+
+    def self.call(assessment_id:, income:, outgoings:)
+      new(assessment_id, income, outgoings).call
+    end
+
+    def initialize(assessment_id, income, outgoings)
+      super()
+      @assessment_id = assessment_id
+      @income = income
+      @outgoings = outgoings
+      @errors = []
+    end
+
+    def call
+      create
+      self
+    end
+
+    def success?
+      @errors.empty?
+    end
+
+    private
+
+    def valid_dates
+      @valid_dates ||= [
+        assessment.submission_date.beginning_of_month - 3.months,
+        assessment.submission_date.beginning_of_month - 2.months,
+        assessment.submission_date.beginning_of_month - 1.month
+      ]
+    end
+
+    def pretty_dates
+      valid_dates.map { |d| d.strftime('%F') }.join(', ')
+    end
+
+    def create
+      [@income, @outgoings].each { |categories| validate_categories(categories) }
+      return unless @errors.empty?
+
+      ActiveRecord::Base.transaction do
+        @income.each { |category_hash| create_category(category_hash, 'credit') }
+        @outgoings.each { |category_hash| create_category(category_hash, 'debit') }
+      rescue StandardError => error
+        @errors << "#{error.class} :: #{error.message}\n#{error.backtrace.join("\n")}"
+      end
+    end
+
+    def validate_categories(categories)
+      categories.each { |category_hash| validate_category(category_hash) }
+    end
+
+    def validate_category(category_hash)
+      if category_hash[:payments].size != 3
+        @errors << "There must be exactly 3 payments for category #{category_hash[:category]}"
+        return
+      end
+      validate_payment_dates(category_hash)
+    end
+
+    def validate_payment_dates(category_hash)
```

Comment:
could #validate_payment_dates be tested, please, especially around `unless dates == valid_dates `

Reply:
It is tested here: https://github.com/ministryofjustice/check-financial-eligibility/blob/c4283889e259bbe1117b162eb2f4d5bae35d6ba4/spec/services/creators/cash_transactions_creator_spec.rb#L75